### ISSUE TITLE
Changing labels while editing issue kills your changes.

### ISSUE THREAD

**Issue by Jaffex** [go-gitea/gitea#8342]
- Gitea version (or commit ref): 1.10.0+dev-352-g7ff783b73
(Tested on try.gitea.io)

- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (https://try.gitea.io/jaffex/test/issues/1)
  - [ ] No
  - [ ] Not relevant

## Description

If you change the labels of an issue while you're editing its text, your changes will disappear after closing the label-selection dialogue.

## Screenshots

![example](https://user-images.githubusercontent.com/31133207/65973708-13205e00-e46c-11e9-813c-5477773e3dd7.gif)

**Comment by bagasme** [comment]
Please give more details (Git version on your server, your OS, etc).

@lunny shouldn't this issue a bug?

**Comment by Jaffex** [comment]
On my own server, here's the specs:

- Gitea version (or commit ref): 1.9.0
- Git version: 2.7.4
- Operating system: Ubuntu 16.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite

**Comment by blueworrybear** [comment]
Hi, it seems to me that the issue is caused by  `function initListSubmits(selector, outerSelector)`. 

As codes below show, the web page is forced to reload each time the `dropdown` menu hide. I think it's because **assignee** and **milestone** are render in server side. To update those information, the original codes force to reload the whole page.

```js
$('.' + selector).dropdown('setting', 'onHide', function(){
    hasLabelUpdateAction = $listMenu.data('action') == 'update'; // Update the var
    if (hasLabelUpdateAction) {
        const promises = [];
        Object.keys(labels).forEach(function(elementId) {
            const label = labels[elementId];
            const promise = updateIssuesMeta(
                label["update-url"],
                label["action"],
                label["issue-id"],
                elementId
            );
            promises.push(promise);
        });
        Promise.all(promises).then(reload);
    }
});
```

Since all **labels** are loaded to web page by default and could be shown/hided with **JavaScript**, it's not necessary to reload. Therefore there maybe two options to fix this issue:

1. Prevent reloading if `selector` equals to **select-label**
    > Although this could quickly fix the issue for updating labels, updating **milestone** and **assignee** would still kill editing content.

2. Move some server-side render content to front-end. (This may take more effort to do so 😅 )

Please let me know what you guys think. I could try to fix this issue. 😄

**Comment by brupelo** [comment]
Related issue: https://github.com/go-gitea/gitea/issues/11004

**Comment by lunny** [comment]
This should be resolved by #23876
